File: src/ManagementPlatform.Domain/Invoice.cs
namespace ManagementPlatform.Domain;

public sealed class Invoice
{
    public long Id { get; set; }
    public long CheckoutAttemptId { get; set; }
    public CheckoutAttempt CheckoutAttempt { get; set; } = null!;
    public InvoiceStatus Status { get; set; } = InvoiceStatus.Pending;
    public string? FailureReason { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset? CompletedAt { get; set; }
}

File: src/main/java/com/managementplatform/domain/model/Invoice.java
package com.managementplatform.domain.model;

import com.managementplatform.domain.enums.InvoiceStatus;
import java.time.Instant;

public record Invoice(
    long id,
    long checkoutAttemptId,
    InvoiceStatus status,
    String failureReason,
    Instant createdAt,
    Instant completedAt
) {
}
